### DISCUSSION TITLE
device_internet_access no longer available

### DISCUSSION THREAD

**Discussion by dominicusmento** [Vaskivskyi/ha-asusrouter#672] [Q&A] [answered]
Can somebody write why this is no longer available?
https://asusrouter.vaskivskyi.com/features/parental-control.html#device-internet-access

**Comment by Vaskivskyi** [comment, marked as answer]
## Short answer

This is a temporary issue due to migration to the new backend library. The possibility to control internet access per-device will be added back in the next release in a new form. The current implementation was no longer compatible.

---

## Long answer (copied from [here](https://community.home-assistant.io/t/custom-component-asusrouter-integration/416111/633?u=vaskivskyi))

### :open_book: A story of AsusRouter services

As you have noticed, in the latest beta releases `0.23.0bX` all the sensors earlier available in AsusRouter were removed. This is not a final decision to remove some functionality, but rather to adapt it to the new ways Home Assistant can work with integrations as well as to the new backend library `AsusRouter`.

Before, AsusRouter had the following services:
- `adjust_wlan` to change settings of WLAN and Guest WLAN networks
- `device_internet_access` for parental control rules
- `remove_trackers` to remove useless device trackers from HA
- `port_forwarding` for port forwarding rules

The plan is to revive them in either a similar or a bit different form. E.g. by the possibility of creating actual switches for parental control rules, that can be easily automated by users in HA automations or easily added to the frontend cards.
